lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -1115,6 +1213,42 @@ SubExprInfo ExprEmitter::visitComb(MuxOp op) {
   return {Conditional, signedness};
 }
 
+SubExprInfo ExprEmitter::visitTypeOp(StructCreateOp op) {
+  StructType stype = op.getType().cast<StructType>();
+  os << "'{";
+  size_t i = 0;
+  llvm::interleaveComma(stype.getElements(), os,
+                        [&](const StructType::FieldInfo &field) {
+                          os << field.name << ": ";
+                          emitSubExpr(op.getOperand(i++), Selection);
+                        });
+  os << '}';
+  return {Unary, IsUnsigned};
```

Comment:
It would be fine to split this off to a separate patch, but I think it would make sense to add an `IsStruct` case to `SubExprSignResult`.